### DISCUSSION TITLE
Unable to download from youtube: "Sign in to confirm you’re not a bot"

### DISCUSSION THREAD

**Discussion by anthonyhtang** [mxpv/podsync#725] [General]
podsync suddenly stopped working since last week, I checked the log file and found the error message "Sign in to confirm you’re not a bot, etc. etc.". 
The reason is likely to be that YouTube recently blocked yt-dlp, and a possible solution, according to [this issue](https://github.com/yt-dlp/yt-dlp/issues/10128), is not passing the cookies. Could you please help update podsync accordingly? Thank you very much!

**Comment by AlrclyGit** [comment]
I'm having this problem too.

**Comment by Gasparos** [comment]
Same here

**Comment by steav** [comment]
Same for me, looks like Hetzer Cloud + Server IPs got blocked by YT (at least for me).
Workaround: smartproxy.com pay-as-you-go. Podsync config: `youtube_dl_args = ["-vU", "--proxy", "socks5h://YOURCREDENTIALS@gate.smartproxy.com:7000"]` => works for me.

**Comment by Elycoo** [comment]
I also have the same problem.
The point podsync uses a token from my account. Isn't it considered a "log-in"?

I'm trying to disable requests for some time to youtube. I will try again in 3 days.

**Comment by Elycoo** [comment]
> Same for me, looks like Hetzer Cloud + Server IPs got blocked by YT (at least for me). Workaround: smartproxy.com pay-as-you-go. Podsync config: `youtube_dl_args = ["-vU", "--proxy", "socks5h://YOURCREDENTIALS@gate.smartproxy.com:7000"]` => works for me.

@steav Which plan do I need to choose?
I don't understand the differences.. Too complicated for me..



> I also have the same problem. The point podsync uses a token from my account. Isn't it considered a "log-in"?
> 
> I'm trying to disable requests for some time to youtube. I will try again in 3 days.

It doesn't seem to work after all :(

**Comment by Harnas** [comment]
Podsync uses yt-dlp to download from youtube. Last time, Youtube changes their policy, and now yt-dlp have problems with downloading. More about that https://github.com/yt-dlp/yt-dlp/issues/10128

**Comment by Elycoo** [comment]
I used Oauth2 to log in with a new youtube account. I had to use cookies too. And to specify the cache-dir for yt-dlp too.
It was quite hard, but now it works again.
Good luck!

**Comment by d4os** [comment]
I moved podsync to my home PC, since I keep it on 24/7. You could also setup a VPN host at home and make your podsync server use it. It'd just keep retrying when it's off, but pull things when the vpn is available.

**Comment by Harnas** [comment]
@d4os try use yt-dlp nightly and set PO token https://github.com/yt-dlp/yt-dlp/issues/10128#issuecomment-2350782316 

For each feed in file config, You must add following parameters to "youtube_dl_args"
`youtube_dl_args = [ .. "--sleep-requests", "1.5", "--min-sleep-interval", "60", "--max-sleep-interval", "90", "--cookies", "/app/cookies.txt", "--extractor-args", "youtube:player-client=web,default;po_token=web+ABCD...FGH"]`

**Comment by d4os** [comment]
@Harnas do you need info if that worked? My clunky solution works, so I prefer it to something that

> may help in some cases

So many websites outright block my Hetzner server that I accepted I need to move some things home.

**Comment by halllo** [comment]
I have the same problem. I dont think it has to do with a blocklisted IP, because everyday I get a new IP.

**Comment by halllo** [comment]
I dont understand how to get the PO token and the cookies. I followed [this guide](https://github.com/yt-dlp/yt-dlp/wiki/Extractors#po-token-guide) but to no avail. My requests dont seem to have them.

Did anybody get it working correctly and reliably?

**Comment by jerryhanson** [comment]
Same problem here, but successfully tackled by adding this line for each feed.
  youtube_dl_args = [ "--cookies", "/app/cookies.txt"]
Cookie file for the youtube website was extracted through Chrome extension, "Get cookies.txt locally", search it in the store. For more info about this, you can refer to Metube github page.

**Comment by brothamanben** [comment]
> Same problem here, but successfully tackled by adding this line for each feed. youtube_dl_args = [ "--cookies", "/app/cookies.txt"] Cookie file for the youtube website was extracted through Chrome extension, "Get cookies.txt locally", search it in the store. For more info about this, you can refer to Metube github page.

Can you show me a sample of your config.toml file? I'm not sure exactly where to add this

**Comment by halllo** [comment]
Do you have to manually update the cookies.txt every day?

**Comment by brothamanben** [comment]
> Do you have to manually update the cookies.txt every day?

I have no idea, it's not working for me yet

**Comment by anthonyhtang** [comment]
> [@Harnas](https://github.com/Harnas) do you need info if that worked? My clunky solution works, so I prefer it to something that
> 
> > may help in some cases
> 
> So many websites outright block my Hetzner server that I accepted I need to move some things home.

Tried racknerd and lisahost, the same problem.

**Comment by brothamanben** [comment]
> [@d4os](https://github.com/d4os) try use yt-dlp nightly and set PO token [yt-dlp/yt-dlp#10128 (comment)](https://github.com/yt-dlp/yt-dlp/issues/10128#issuecomment-2350782316)
> 
> For each feed in file config, You must add following parameters to "youtube_dl_args" `youtube_dl_args = [ .. "--sleep-requests", "1.5", "--min-sleep-interval", "60", "--max-sleep-interval", "90", "--cookies", "/app/cookies.txt", "--extractor-args", "youtube:player-client=web,default;po_token=web+ABCD...FGH"]`

this stilll didn't work for me, is there something else I can do?

**Comment by d4os** [comment]
> > [@d4os](https://github.com/d4os) try use yt-dlp nightly and set PO token [yt-dlp/yt-dlp#10128 (comment)](https://github.com/yt-dlp/yt-dlp/issues/10128#issuecomment-2350782316)
> > For each feed in file config, You must add following parameters to "youtube_dl_args" `youtube_dl_args = [ .. "--sleep-requests", "1.5", "--min-sleep-interval", "60", "--max-sleep-interval", "90", "--cookies", "/app/cookies.txt", "--extractor-args", "youtube:player-client=web,default;po_token=web+ABCD...FGH"]`
> 
> this stilll didn't work for me, is there something else I can do?

I am using ` yt-dlp version stable@2025.01.15 from yt-dlp/yt-dlp [c8541f8b1]`
A cookie I generated in a private window at home
With a proxy so that the download is made from my home address
It is working. If you're trying to download from an address that works in the browser and have that cookie, maybe your yt-dlp is outdated? I had to run `yt-dlp -U` because it stopped working a couple days ago.
I suggest narrowing the investigation to testing out a yt-dlp command, once it's working, making Podsync use the same arguments is easy.

**Comment by sunrisewestern** [comment]
> > > [@d4os](https://github.com/d4os) try use yt-dlp nightly and set PO token [yt-dlp/yt-dlp#10128 (comment)](https://github.com/yt-dlp/yt-dlp/issues/10128#issuecomment-2350782316)
> > > For each feed in file config, You must add following parameters to "youtube_dl_args" `youtube_dl_args = [ .. "--sleep-requests", "1.5", "--min-sleep-interval", "60", "--max-sleep-interval", "90", "--cookies", "/app/cookies.txt", "--extractor-args", "youtube:player-client=web,default;po_token=web+ABCD...FGH"]`
> > 
> > 
> > this stilll didn't work for me, is there something else I can do?
> 
> I am using ` yt-dlp version stable@2025.01.15 from yt-dlp/yt-dlp [c8541f8b1]` A cookie I generated in a private window at home With a proxy so that the download is made from my home address It is working. If you're trying to download from an address that works in the browser and have that cookie, maybe your yt-dlp is outdated? I had to run `yt-dlp -U` because it stopped working a couple days ago. I suggest narrowing the investigation to testing out a yt-dlp command, once it's working, making Podsync use the same arguments is easy.

I guess it is the proxy that make it work...

**Comment by rodhfr** [comment]
The docker is using some old yt-dlp version
I fixed mine with `docker exec -it podsync /usr/bin/youtube-dl -U` for current instance but apparently auto-update as stated in the config doesn't work properly?.
```toml
[downloader]
# Optional, auto update youtube-dl every 24 hours
self_update = true
# Download timeout in minutes.
timeout = 15
```
```bash
[rodhfr@fedora ~]$ docker exec -it podsync /usr/bin/youtube-dl -U
Current version: stable@2024.03.10 from yt-dlp/yt-dlp
Latest version: stable@2025.03.31 from yt-dlp/yt-dlp
Current Build Hash: 5576348930e44102add235c0a032e766f416a507db1dd2c75da833ff21e9fb6c
Updating to stable@2025.03.31 from yt-dlp/yt-dlp ...
Updated yt-dlp to stable@2025.03.31 from yt-dlp/yt-dlp
```
